src/proteingym/base/splits.py

```diff
@@ -102,6 +123,16 @@ def _reshape_list(flat_list: list, shape: tuple[int, ...]) -> list:
     return reshaped_list
 
 
+def _unique_sequences_for_targets(
+    dataset: Dataset, targets: list[str] = None
+) -> list[Seq]:
+    sequences = set()
+    for assay in dataset.assays:
+        if not targets or (targets and any(t in assay.columns for t in targets)):
```

Comment:
So the intent is to see if the targets are in the assay and not that the targets contain `None` values. The later would be more strict